src/test/suite/editors/playgroundController.test.ts

```diff
@@ -330,5 +354,26 @@ suite('Playground Controller Test Suite', () => {
         }
       ).then(done, done);
     });
+
+    test('cancel a playground', (done) => {
+      const mockDocument = {
+        _id: new ObjectId('5e32b4d67bf47f4525f2f729'),
+        field: 'sample'
+      };
+
+      seedDataAndCreateDataService('forest', [mockDocument]).then(
+        async (dataService) => {
+          testConnectionController.setActiveConnection(dataService);
+
+          testLanguageServerController.executeAll('while (1===1) {}', 'mongodb://localhost:27018');
+
+          await testLanguageServerController.cancelAll();
+
+          const result = await testLanguageServerController.executeAll('4 + 4', 'mongodb://localhost:27018');
+
+          expect(result).to.be.equal('8');
```

Comment:
It is more about real-life testing. We had a call with Lucas and agreed that we omit manual connections closing since terminating the entire thread should do the job. But we created a ticket as kind of a reminder to review it later. This ticket can be closed in the future if no problem appears. The more complex tests can be implemented after emerging async rewriting in `mongosh`. Currently, when a playground has several commands they can be executed in the random sequence what leads to unexpected results. See: https://mongodb.slack.com/archives/GTTUKDWRH/p1586851227003500